src/sagas.js

```diff
@@ -389,6 +394,17 @@ function* schedulerPerMinute (action) {
   }
 }
 
+function* fetchPermissionWithoutFilter (action) {
+  const data = yield callExternalAction('checkFilter', Api.checkFilter, { action: 'CHECK_FILTER' })
+  yield put(changeFilterPermissions(data.error !== undefined))
+  yield put(setAdministrator(data.error === undefined))
+}
+
+function* changeUserFilterPermission (action) {
+  yield put(setUserFilterPermission(action.payload.filter))
+  yield put(getAllVms({ shallowFetch: false }))
```

Comment:
As I wrote above, getAllVms must be called only after it check user role, for me there only two ways, first make delay (which is the worst). and second is make it nested. __changeUserFilterPermission__ is only use in logining and changing filter, so I think make flow little more complex for it, is not so bad solution